src/mesher/Mesher.cpp

```diff
@@ -0,0 +1,77 @@
+#include "gmsh.h"
+#include "Mesher.h"
+#include <string>
+
+
+Mesher::Mesher() { gmsh::initialize(); }
+Mesher::~Mesher(){ gmsh::finalize(); }
```

Comment:
We need to think about what happens when a `Mesher` instance is moved or copied:

If you copy a `Mesher` instance, will `gmsh::initialize` be called twice? Can gmsh handle this and what does this entail? What happens if you destruct one `Mesher` copy, effectively calling `gmsh::finalize`? Will the other copy of the mesher still work?

Reply:
We should also take care about what happens if there are two `Mesher` instances. 

 - Does gmsh use global variables somehow? If yes, the two mesher instances would interfere with each other.
 - Once the first goes out of scope, `gmsh::finalize()` will be called, while the second one is still in use. 

If it is true, that there can never be more than one `Mesher` instance, we should reflect this in the code design, e.g. by implementing a so-called Meyer's singleton: https://godbolt.org/z/MKdrfGTo7

Reply:
Good point. I'll implement the Meyer's singleton.

Reply:
I thought about this, and I now believe a Meyer's singleton is not enough: The Meyer's singleton might make sure that we call `gmsh::initialize()` and `gmsh::finalize()` only once, but it does not safeguard from two portions of the code using the same singleton mesher to mesh different parts requiring different options. They would still interfere with each other. I can't think of a good programming pattern to prevent this off the top of my head, so let's revert the Meyers singleton stuff for now, so that we don't overengineer and just keep in mind that our Mesher will not be threadsafe.